clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -626,7 +631,8 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                         "Stream interface version != trace marker");
     }
     if (memref.marker.type == TRACE_TYPE_MARKER &&
-        memref.marker.marker_type == TRACE_MARKER_TYPE_CHUNK_FOOTER) {
+        memref.marker.marker_type == TRACE_MARKER_TYPE_CHUNK_FOOTER &&
+        !is_dynamically_core_sharded(shard)) {
```

Comment:
Isn't this more like the single-thread-state checks? This applies to input streams, not the output stream.  So when the input != output (dynamically core sharded), we would want to keep per-input == per-thread state and apply these same checks within that state: much like all the thread state like continuity and signals.

So this should be treated the same way instead of having a separate kind of check, right?  What are you doing for the thread state: you're resetting/clearing on a switch? The same thing would work here, to reset the last_chunk_ordinal_, and when we have per-input state we can move that field into that state.

Reply:
Longer-running state like retaddr_stack_, last_signal_context_, and signal_stack_ are all reset. Setting them to empty works fine because the subsequent checks behave as if we're at start of trace and there wasn't a prior signal context. But the chunk ordinal check is stricter in that sense; it expects `last_chunk_ordinal_+1` always, so it's unable to "just work".